src/plugins/kibana_utils/server/report_server_error.ts

```diff
@@ -43,6 +56,7 @@ export function reportServerError(res: KibanaResponseFactory, err: KbnServerErro
     body: {
       message: err.message,
       attributes: err.errBody?.error,
+      ...(err.requestParams ? { requestParams: err.requestParams } : {}),
```

Comment:
`KbnServerError` and `reportServerError` are widely used outside of data plugin for catching any kind of error in route handlers. It's risky to return the 'requestParams' for all the places this gets used and this would also not be a pattern that we'd recommend gets widely adopted.

We should rather move this functionality and `sanitizeRequestParams` into the data plugin.

Reply:
I am not sure I understand, are you suggesting not using KbnServerError and reportServerError in data plugin?

Reply:
yes, we could create a new `reportElasticErrorAndRequestParams` (naming tbd) in the data plugin and leave the `reportServerError` util as-is so we don't change the APIs of any other plugins/teams.

Reply:
ok, I will do that. That aligns well with another effort to return rawResponse from data plugin search endpoints, https://github.com/elastic/kibana/issues/167099. I will work https://github.com/elastic/kibana/issues/167099 first to migrate data search plugin away from KbnServerError and reportServerError so that rawResponse is included in error response. Then I can circle back to this PR and build off of those data plugin only utility methods